src/concurrency/sync.rs

```diff
@@ -731,9 +731,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                     let futex = this.machine.sync.futexes.get_mut(&addr).unwrap();
                     futex.waiters.retain(|waiter| waiter.thread != thread);
                     // Set errno and write return value.
-                    this.set_last_error(errno_timeout)?;
-                    this.write_scalar(retval_timeout, &dest)?;
```

Comment:
This didn't write `-1`, it wrote `retval_timeout`... so your patch changes behavior, and now `retval_timeout` is entirely ignored?

Please revert this part. And generally be more careful about this -- don't touch code you don't understand.

Reply:
I see, I understand why this is wrong. I'll be more careful. Thank you for the close reviews again.